packages/babel-generator/src/generators/typescript.ts

```diff
@@ -498,6 +498,27 @@ function tokenIfPlusMinus(self: Printer, tok: true | "+" | "-") {
   }
 }
 
+export function TSTemplateLiteralType(
+  this: Printer,
+  node: t.TSTemplateLiteralType,
+) {
+  const quasis = node.quasis;
+
+  let partRaw = "`";
+
+  for (let i = 0; i < quasis.length; i++) {
+    partRaw += quasis[i].value.raw;
+
+    if (i + 1 < quasis.length) {
+      this.token(partRaw + "${", true);
+      this.print(node.types[i]);
+      partRaw = "}";
+    }
+  }
+
+  this.token(partRaw + "`", true);
+}
```

Comment:
We can extract this to a function `_printTemplate(quasis, substitutions)`, so that we don't have to keep the same logic in two different printers.